### DISCUSSION TITLE
Theming template repo

### DISCUSSION THREAD

**Discussion by anatolinicolae** [vuejs/vitepress#2561] [General]
I've gone through the docs section about [Distributing a Custom Theme](https://vitepress.dev/guide/custom-theme#distributing-a-custom-theme) as an npm package, but I feel like instructions are pretty generic.

What I'm virtually looking for is an npm distributed theme that's self-sufficient and requires little config on user side. Currently I can surely copy over overrides and component definitions to another project, but it should be reusable in multiple projects with minimal code duplication. So distributing a base custom theme that requires you to only change site name and description would be the best thing to implement this "common core".

Is there any way to work out a template repo that provides a skeleton structure for theming VitePress and distributing it as an npm package? Or is there an example of how to do exactly this?

**Comment by brc-dd** [comment]
https://github.com/vuejs/theme , https://www.npmjs.com/search?q=vitepress%20theme
